### PR TITLE
Move kubectl overview to be section index

### PR DESCRIPTION
**Pull request by sftim**
- [Old page](https://kubernetes.io/docs/reference/kubectl/overview/) (also see its [parent](https://kubernetes.io/docs/reference/kubectl/))
- [New page](https://deploy-preview-29847--kubernetes-io-main-staging.netlify.app/docs/reference/kubectl/)

Also:
- use glossary definition in page introduction
- tidy broken link in What's Next section
- redirect old URL to new URL

/sig cli
/language en
/kind cleanup

### CODE REVIEW

**Comment by sftim** [comment]
An enabling change for https://github.com/kubernetes/website/issues/25943

File: content/en/docs/reference/glossary/kubectl.md [hunk 1]
```diff
@@ -4,16 +4,19 @@ id: kubectl
 date: 2018-04-12
 full_link: /docs/user-guide/kubectl-overview/
 short_description: >
-  A command line tool for communicating with a Kubernetes API server.
+  A command line tool for communicating with a Kubernetes cluster.
```
Comment (mkorbi) [inline content/en/docs/reference/glossary/kubectl.md:7]: A command line tool for communicating with a Kubernetes cluster API server.

File: content/en/docs/reference/kubectl/_index.md [hunk 2]
```diff
@@ -1,5 +1,548 @@
...
+sudo mv ./kubectl-whoami /usr/local/bin
+
+kubectl whoami
+Current user: plugins-user
+```
+
+## {{% heading "whatsnext" %}}
+
+* Read the `kubectl` [command reference](/docs/reference/kubectl/kubectl/).
+
```
Comment (mkorbi) [inline content/en/docs/reference/kubectl/_index.md:547]: Find some further useful command examples on the `kubectl` [cheat sheet](/docs/reference/kubectl/cheatsheet/)

File: content/en/docs/reference/glossary/kubectl.md (hunk 1 above)
Comment (mkorbi) [reply content/en/docs/reference/glossary/kubectl.md:7]: for me both is right depending on the target group

File: content/en/docs/reference/kubectl/_index.md (hunk 2 above)
Comment (sftim) [reply content/en/docs/reference/kubectl/_index.md:547]: I usually prefer to keep migration PRs separate from PRs that add new content.

**Comment by jimangel** [comment]
/lgtm

I wonder if this should land after the release and if we need to boost visibility to localization (to me it feels like a small big change vs. a refactor)?

**Comment by sftim** [comment]
Happy to let this land after the v1.23 release and yes, we should consider applying the refactor label.
/label refactor

**Comment by sftim** [comment]
@jimangel the only changes since your previous LGTM were a (quite big) rebase with no manual rework

File: content/en/docs/reference/glossary/kubectl.md (hunk 1 above)
Comment (kbhawkey) [reply content/en/docs/reference/glossary/kubectl.md:7]: Calling out, "Kubernetes API server" seems more specific than "cluster". Is it more important to identify the Kubernetes API server?

File: content/en/docs/reference/kubectl/_index.md [hunk 3]
```diff
@@ -1,5 +1,551 @@
...
+sudo mv ./kubectl-whoami /usr/local/bin
+
+kubectl whoami
+Current user: plugins-user
+```
+
+## {{% heading "whatsnext" %}}
+
+* Read the `kubectl` [command reference](/docs/reference/kubectl/kubectl/).
+* Read the `kubectl` [command line arguments](/docs/reference/kubectl/kubectl/) reference.
```
Comment (kbhawkey) [inline content/en/docs/reference/kubectl/_index.md:549]: 👋 @sftim, Can you check the links for the first two items (links are the same)?
The page could be configured to ignore the list of section links at the bottom of the page, or do you think the links add value?

File: content/en/docs/reference/kubectl/_index.md [hunk 4]
```diff
@@ -1,5 +1,551 @@
 ---
-title: "kubectl"
+title: Command line tool (kubectl)
```
Comment (kbhawkey) [inline content/en/docs/reference/kubectl/_index.md:2]: nit: Should the title contain Kubernetes?
Kubernetes command-line tool (kubectl)

File: content/en/docs/reference/glossary/kubectl.md (hunk 1 above)
Comment (sftim) [reply content/en/docs/reference/glossary/kubectl.md:7]: Here's why I prefer “cluster”: if we say “API server” in the tooltip, I'm worried that some newcomers will feel that they need to learn what a Kubernetes API server is&mdash;not an easy thing to grasp&mdash;and that some proportion of those will just write off Kubernetes as too hard to learn.

The full description is welcome to make the details clear.

File: content/en/docs/reference/kubectl/_index.md (hunk 4 above)
Comment (sftim) [reply content/en/docs/reference/kubectl/_index.md:2]: @kbhawkey let's save that for a different PR; that would be a change beyond the page move that this PR's title covers.

File: content/en/docs/reference/kubectl/_index.md (hunk 3 above)
Comment (sftim) [reply content/en/docs/reference/kubectl/_index.md:549]: Good spot.

**Pull request merged**